core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -472,6 +472,47 @@ class DAGScheduler(
   }
 
   /**
+   * Get ancestor splits in ShuffledRDD.
+   */
+  private[spark] def parentSplitsInShuffledRDD(stageId: Int, pId: Int): Option[Map[Int, Set[Int]]] =
```

Comment:
Yes, this is confusing and I need to refine this.
I'm a little bit hesitant to use `getShuffleDependencies`. I need to get the total size of input from `ShuffledRDD` for every child's partition. After transformations like `CoalescedRDD`, there may be not a consistent one-to-one match between ancestor's partition index and child's partition index.